script/GameObjects/Elements/Components/Computer.lua

```diff
@@ -0,0 +1,77 @@
+local Entity = require('GameObjects.Entity')
+local BasicShapes  = require('Systems.Gen.ShapeLib.BasicShapes')
+local SocketType = require('GameObjects.Entities.Ship.SocketType')
+
+local shared
+local rng = RNG.FromTime()
+
+local Computer
+Computer = subclass(Entity, function (self)
```

Comment:
why not ```local Computer =``` right away as everywhere else?

Reply:
The instance is referenced outside its internal scope. When doing the initialization without the separate local declaration, the reference fails. It's a piece of weirdness associated with having to give components going into plugs a simple RigidBody or else their Event functions don't work any more.

Reply:
And by "fails," I mean generates a hard, game-crashing error when the register() call that sends the components Event.Update or Event.Debug function to the Event processor yields a fatal error:

`self:register(Event.Update, Capacitor.updateCapacitor)`

Reply:
use self:updateCapacitor not the local variable

Reply:
In my defense, making the class declaration its own statement is how Josh implemented both Thruster and Turret. My new components were based off of those.

Having said that, I've modified all of the components to use "self" rather than their class name.